### ISSUE TITLE
DT window is too wide

### ISSUE THREAD

**Issue by lethosor** [splintermind/Dwarf-Therapist#232]
The DT window is a good 100px too wide for my screen (1280x800) and can only be decreased in width by 10px or so.
OS X 10.9, DT 30.1.0

**Comment by txtsd** [comment]
Minimum width is wider than my 1280x800 screen in most use cases.
Fits perfectly before connecting to DF.
Using a QT5 compiled version.

**Comment by Unknown-Zombie** [comment]
It's because the Customizations dock refuses to be shrunk smaller than a certain size. If you move the Customizations dock somewhere else, or close it, then the DT window can be shrunk to a size that is usable on a 1280x800 screen.

**Comment by lethosor** [comment]
@splintermind that change works until the "can't connect to DF" dialog is displayed, at which point the DT  window becomes too wide again.
Also, I believe @txtsd is using Linux.

**Comment by fricy** [comment]
a92d294 doesn't fix on 10.7.5 (QT4)

**Comment by Unknown-Zombie** [comment]
I am also using Linux (QT5).

Having any Dock locked to the side of the Main Window causes the program to refuse to be shrunk below a certain width, the sum of which is greater than 1280. Closing all Docks, or moving the Docks to anchor at the top or bottom of the Main Window will allow the window to fit onto a 1280-wide screen.

**Comment by txtsd** [comment]
Yes, I'm on linux so this isn't OSX exclusive.

**Comment by fricy** [comment]
Reopen please, still present in 32.0

**Comment by splintermind** [comment]
Are you kidding me... what resolution are you using? Minimum width should be WAY lower than 1280 now.

**Comment by fricy** [comment]
1280x800. Upon startup the window resizes the grid to show every labor from mining-»mechanics, plus the dock, this is like +400 px. What fits on my screen with this resolution is minig-»tanning+dock, mining-strand extraction without any dock window. Every time I open or a close a dock window DT resizes itself again, which is a bit annoying.
User settings were cleared for testing.

**Comment by splintermind** [comment]
So when you start it up initially, it's already 400px wider than your resolution? When you say 'open or close a dock' what exactly do you mean? Are you moving them around, showing/hiding, and/or docking them to different sides? Are you running in full screen?

**Comment by kraila** [comment]
I'm also having this problem on OSX, resolution 1440x900. I removed tabs from the right (Emotions, Details, Pending Changes) but was still unable to make the window small enough to show less than 1440px wide. I will keep experimenting to see if I find a workaround.

**Comment by splintermind** [comment]
On OSX someone reported that they were unable to move the 2nd toolbar below the first one (the toolbar with group by, filters and population counts).

Can anyone confirm if the toolbar can be moved and/or if it resolves the issue?

**Comment by fnky** [comment]
I can move the toolbars (docks)—I also have the same problem as @KellyTeresa

I suggest the minimum window size is less than or equal to 800x600 or 1024x768 to support a wider range of resolutions.

**Comment by splintermind** [comment]
The dock widgets are not the same as the toolbars. The toolbars are the strips where the buttons are. I believe on OSX the issue is that those toolbars can't be resized or moved and prevent the main widget from being resized below a certain width.

The size of the main window should be able to be reduced quite small and there's something going on with OSX preventing this. On windows, with docks on both sides, it's still possible to reduce the size to ~700x500.

**Comment by fnky** [comment]
I see what you mean. The toolbar in OSX is very fluid. I don't know how Qt handles this, but natively, items will be squished into a menu:

<img width="432" alt="screen shot 2016-01-24 at 14 11 48" src="https://cloud.githubusercontent.com/assets/995050/12536416/9f59bc06-c2a4-11e5-92fe-163f4141d776.png">

**Comment by splintermind** [comment]
That's exactly what it should do in DT, but perhaps it's not working on OSX?
![](http://i.imgur.com/XBWCjwE.jpg)

**Comment by fnky** [comment]
I'll try to figure out a fix for OSX.

**Comment by fnky** [comment]
Nevermind, figured that I was using an older version than DR2016 branch. The toolbar does work as intended on OSX. This was an issue in _32.0.0_ which the download link in the README links to.

Wish this was updated, as the majority of people use that, link without knowing it's outdated. I'd suggest to use [git tags](https://git-scm.com/book/en/v2/Git-Basics-Tagging) and Release functionality in GitHub.

**Comment by lethosor** [comment]
> I'd suggest to use git tags and Release functionality in GitHub.

That's already what splinterz is using (except he just uploads Windows builds): https://github.com/splintermind/Dwarf-Therapist/releases

**Comment by fnky** [comment]
Yeah just saw that; for some reason it didn't who me the link. Is there a reason that there aren't OSX and Linux builds?

**Comment by Hello71** [comment]
because fricy makes them and Linux builds are a stupid idea. respectively.

**Comment by fnky** [comment]
I don't see how that would be “a stupid idea”, not a valid reason in my book.

**Comment by fricy** [comment]
> Is there a reason that there aren't OSX [...] builds?

@splintermind : [Here's a share](https://www.dropbox.com/sh/eacs1ofvcd433io/AADrdaliXwtWgmjqk0RffwSOa?dl=0) of a few builds from my archive for the releases. If you need any more, I can rebuild them, but this covers the final 40.24 build, and all 42.x-s.

**Comment by splintermind** [comment]
@fnky I've updated the 2016 branch DFFD OSX link and changed the readmes in both the new and older branch which will hopefully clarify where releases are.

@fricy Thanks for the packages, I've added them to the last releases for the current DT and the last version for 40.24.
